packages/devextreme/testing/tests/DevExpress.ui.widgets.htmlEditor/htmlEditorParts/aiDialog.tests.js

```diff
@@ -0,0 +1,170 @@
+        });
+
+        this.aiDialog._popup.hide();
+    });
+
+    QUnit.test('Should hide options SelectBox if command with no options selected', function(assert) {
+        showAiDialog(this);
+
+        const commandSelectBox = getCommandSelectBoxInstance(this.$element);
+        const optionSelectBox = getOptionSelectBoxInstance(this.$element);
+
+        assert.strictEqual(optionSelectBox.option('visible'), true, 'option SelectBox is initially visible');
+
+        commandSelectBox.option('value', 'summarize');
+
+        assert.strictEqual(optionSelectBox.option('visible'), false, 'option SelectBox hidden after changing command');
+    });
+
+    QUnit.test('Should show options SelectBox and select first option if command with options selected', function(assert) {
+        showAiDialog(this, { isBasicCommand: false, config: { currentCommand: 'summarize' } });
+
+        const commandSelectBox = getCommandSelectBoxInstance(this.$element);
+        const optionSelectBox = getOptionSelectBoxInstance(this.$element);
+
+        assert.strictEqual(optionSelectBox.option('visible'), false, 'option SelectBox is initially not visible');
+
+        commandSelectBox.option('value', 'translate');
+
+        assert.strictEqual(optionSelectBox.option('visible'), true, 'option SelectBox is visible after changing command');
+        assert.strictEqual(optionSelectBox.option('value'), 'english', 'first command option is selected after command change');
+    });
+
+    ['replace', 'insertAbove', 'insertBelow'].forEach((mode) => {
+        QUnit.test(`Should resolve with correct payload on ${mode} click`, function(assert) {
+            const done = assert.async();
+            const hideSpy = sinon.spy(this.aiDialog, 'hide');
+
+            showAiDialog(this).done((resultText, event) => {
+                assert.strictEqual(resultText, 'Test text', 'resolved text is correct');
+                assert.strictEqual(event.itemData.id, mode, `operation is correct: ${mode}`);
+                assert.strictEqual(hideSpy.calledOnce, true, 'hide called');
+                done();
+            });
+
+            selectInsertionMode(mode);
+        });
+    });
+
+    QUnit.test('Copy button triggers clipboard write', function(assert) {
+        if(!navigator.clipboard) {
+            assert.ok(true, 'clipboard not supported in this environment');
+            return;
+        }
+
+        const clipboardStub = sinon.stub(navigator.clipboard, 'writeText');
+
+        showAiDialog(this);
+
+        const $copyButton = findButtonByText(this.$element, 'Copy');
+        $copyButton.trigger('dxclick');
+
+        assert.strictEqual(clipboardStub.calledWith('Test text'), true, 'copied correct text');
+
+        clipboardStub.restore();
+    });
+
+    QUnit.test('Popup config should contain correct parameters', function(assert) {
+        const popupConfig = this.aiDialog['_getPopupConfig']();
```

Comment:
[minor]
It'd be less synthetic if we get opened popup's `option()` to at least `aiDialog._popup.option()`